packages/@aws-cdk/aws-ec2/lib/vpn.ts

```diff
@@ -252,12 +252,6 @@ export class VpnConnection extends cdk.Resource implements IVpnConnection {
       }
 
       props.tunnelOptions.forEach((options, index) => {
-        if (options.preSharedKey && !/^[a-zA-Z1-9._][a-zA-Z\d._]{7,63}$/.test(options.preSharedKey)) {
```

Comment:
What this needs is a `if (!Token.isUnresolved(optios.preSharedKey))` check in there, and a test.

Reply:
Nice idea. Will try that